### PR TITLE
Publish snapshot Docker image

### PR DESCRIPTION
**Pull request by arugal**

### CODE REVIEW

File: .github/workflows/publish-docker.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: publish-docker
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
```
Comment (kezhenxu94) [inline .github/workflows/publish-docker.yaml:23]: Should not do this in pull request

File: .github/workflows/publish-docker.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+env:
+  SKIP_TEST: true
+  HUB: ghcr.io/apache/skywalking
```
Comment (kezhenxu94) [inline .github/workflows/publish-docker.yaml:27]: Should be deployed to skywalking-agent-test-tool

```suggestion
  HUB: ghcr.io/apache/skywalking-agent-test-tool
```

File: .github/workflows/publish-docker.yaml (hunk 1 above)
Comment (kezhenxu94) [reply .github/workflows/publish-docker.yaml:23]: Also, this would always fail because you're open the PR from a fork repo, we

Comment (arugal) [reply .github/workflows/publish-docker.yaml]: I need to open a new PR?

**Comment by kezhenxu94** [comment]
Just push a test branch , once it passed, we can merge this PR and delete that branch

https://github.com/apache/skywalking-agent-test-tool/runs/3111126500

**Review by kezhenxu94 — APPROVED** [review]
https://github.com/apache/skywalking-agent-test-tool/pkgs/container/skywalking-agent-test-tool%2Fmock-collector

**Pull request merged**